### ISSUE TITLE
Expected time and resources for stratifying human genome splits

### ISSUE THREAD

**Issue by vitkl** [de-Boer-Lab/hashFrag#1]
Hi @muntakimrafi 

This is a really useful approach and package, congrats on getting the paper out!

Could I ask how much time and resources one should expect for stratifying human genome splits?

In particular, 2.4mln of 2000bp/1000bp tiles as training set and ~100k test set. Is this feasible?

I could select a subset of both - so you have ~90k training and ~5k tests (differential across cell types or random) - but the model would have seen 2.4mln during training. If you have to use smaller sets, do you think that is a reasonable approach?

**Comment by muntakimrafi** [comment]
Hello @vitkl, this is feasible but you need to divide the queries into subsets and run hundreds of jobs on HPC. 

I just checked it for 100seqs (queries) compared against 2.4Mseqs (database) [all 1000 bp length] and the runtime was ~40min in my workstation (ran with 14 threads). 

My suggestion would be to use hashFrag blastn_module on subsets of test sequences and later combine all the blastn output files.

**Comment by muntakimrafi** [comment]
we will try to push few code soon to handle large datasets.

**Comment by vitkl** [comment]
Thanks for the suggestions!

Do you mean that instead of running this
```bash
hashFrag stratify_test_split \
--train_fasta_path example_train_split.fa.gz \
--test_fasta_path example_test_split.fa.gz \
-o stratify_test_split.work
```
for 100s of `example_test_split.fa.gz` files each with ~100 sequences, you run `blastn_module`, then aggregate outputs before running `stratify_test_split_module`? 
A code example to do this would indeed be quite helpful. If you plan to add this within a few weeks, I can wait.

Is there a problem with running this
```bash
hashFrag stratify_test_split \
--train_fasta_path example_train_split.fa.gz \
--test_fasta_path example_test_split.fa.gz \
-o stratify_test_split.work
```
for 100s of `example_test_split.fa.gz` files each with ~100 sequences, aggregating scores after?

**Comment by muntakimrafi** [comment]
yes exactly. we should be able to push it soon.

**Comment by muntakimrafi** [comment]
Hello Vitali, we have a new module pushed to the repo. Check it out.
